### ISSUE TITLE
terraform graph doesn't expand at all in 0.4.2 and master

### ISSUE THREAD

**Issue by stefancocora** [hashicorp/terraform#1670]
Like the title says terraform
```bash
terraform version
Terraform v0.4.2
```
and master
```bash
terraform version
Terraform v0.5.0-dev (8c3fba6d24a89aaa79fe7079726a7f3202d6f08e)
```
don't expand the terraform graph when `-module-depth=-1` is set, see here
https://github.com/stefancocora/tf_sg_deps_bug/blob/master/graph.png

terraform 0.3.7 expands the graph properly, see here
https://github.com/stefancocora/tf_sg_deps_bug/blob/master/graph-0.3.7.png

**Comment by phinze** [comment]
Stay tuned! This will be fixed in 0.5.0 with #1661

**Comment by stefancocora** [comment]
Brilliant ! 
Thanks for the context around #1661 
I'll watch closely :)

**Comment by phinze** [comment]
Landed the PR in master, so closing this issue. Feel free to build master and take it for a spin.

**Comment by stefancocora** [comment]
Fast feedback loop coming :)

Unless I'm doing something amazingly wrong it looks like #1661 doesn't fix this bug:

I've just rebuilt TF after I've seen your github post:
```bash
terraform version
Terraform v0.5.0-dev (1061bbfbfe81da8495694b92d8ef2d5045d02fc5)
```

**Generating the graph with TF master** gives me still the unexpanded graph
```bash
terraform version && bash -x ./tf.sh graph
Terraform v0.5.0-dev (1061bbfbfe81da8495694b92d8ef2d5045d02fc5)

+ COMMAND=graph
+ PLAN=plan.tf
+ GRAPH=graph.png
+ [[ graph = \a\p\p\l\y ]]
+ [[ graph = \p\l\a\n ]]
+ [[ graph = \p\l\a\n\d\e\s\t\r\o\y ]]
+ [[ graph = \d\e\s\t\r\o\y ]]
+ [[ graph = \d\e\s\t\r\o\y\n\o\w ]]
+ [[ graph = \u\p\d\a\t\e ]]
+ [[ graph = \g\r\a\p\h ]]
+ echo 'outputting infrastructure graph to graph.png'
outputting infrastructure graph to graph.png
+ terraform graph -module-depth=-1 plan.tf
+ dot -Tpng
```
![graph](https://cloud.githubusercontent.com/assets/281241/7422596/0c777f32-ef86-11e4-9a0f-d03011d20424.png)


Generating the graph with `0.3.7` gives me the expanded graph
```bash
docker run -v $PWD:/data  uzyexe/terraform:v0.3.7 graph -module-depth=-1 plan.tf | dot -Tpng > ./graph-0.3.7.png
```

![graph-0 3 7](https://cloud.githubusercontent.com/assets/281241/7422619/2cf6b3ae-ef86-11e4-8f3e-5c54d3d9fb01.png)

**Comment by phinze** [comment]
Thanks for the fast test!

We did actually lose functionality in the big refactor that landed with 0.4.0 - note that #1661 updated the `-help` to:

```
Usage: terraform graph [options] [DIR]

  Outputs the visual dependency graph of Terraform resources according to
  configuration files in DIR (or the current directory if omitted).
```

It looks like you're testing with an argument of `plan.tf` - is that a planfile? At this point the planfile graph generation needs to be reworked onto the new graph structure.

tl;dr - Try omitting the argument or making the argument a directory.

In fact, when I hand a single file to the command I get an error like this:

```
terraform graph -module-depth=-1 main.tf
Error loading Terraform: Error loading config: configuration path must be a directory: main.tf
```

Here's what the new-style graphs look like when expanded:

```
terraform graph -module-depth=-1
```

![output](https://cloud.githubusercontent.com/assets/37534/7422923/a0a753e2-ef56-11e4-87b8-2bc164b05c03.png)

**Comment by stefancocora** [comment]
I did another `git pull` and build of terraform because `Terraform v0.5.0-dev (1061bbfbfe81da8495694b92d8ef2d5045d02fc5)` wasn't doing what you've posted :)

At the current version 
`Terraform v0.5.0-dev (e3616d391ac01503094621b316921bdd4833e369)`

terraform expands the graph correctly using either:
- a plan file
